### PR TITLE
Implement proof username entry UI

### PR DESCRIPTION
**Pull request by chromakode**
:eyeglasses: @keybase/react-hackers 

Awaiting art for the platform icon badging. There's a note about some hackiness around the `<Input>` component. I felt this was outside scope and ticketed the visual tweak separately.

### CODE REVIEW

File: shared/profile/prove-enter-username.desktop.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
+// @flow
+import React, {Component} from 'react'
+import {Box, Icon, Text, Button, Input, PlatformIcon} from '../common-adapters'
+import {globalStyles, globalColors, globalMargins} from '../styles/style-guide'
+import type {Platforms} from '../constants/types/more'
+import type {Props} from './prove-enter-username'
+
+const platformText : {[key: Platforms | 'btc']: {headerText: string, floatingLabelText?: string, hintText?: string}} = {
+  'twitter': {
```
Comment (chrisnojima) [inline shared/profile/prove-enter-username.desktop.js]: can't we make a little function that generates the common case here?

```
const makeCommon = (platform) => ({
headerText: `Prove your ${platform} identity`,
floatingLabelText: `Your ${platform} username`,
})
```

then just

```
'twitter': { ...makeCommon('Twitter') }, 
```

File: shared/profile/prove-enter-username.desktop.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    }
+
+    return (
+      <Box style={styleContainer}>
+        <Icon style={styleClose} type='iconfont-close' onClick={this.props.onCancel} />
+        <Text type='Header' style={{marginBottom: globalMargins.medium}}>{headerText}</Text>
+        {/* FIXME: awaiting blank icon overlay art here */}
+        <PlatformIcon platform={this.props.platform} overlay={'iconfont-proof-pending'} overlayColor={globalColors.grey} size={48} />
+        <Input style={styleInput} {...inputSizeFix} floatingLabelText={floatingLabelText} hintText={hintText} value={this.state.username} onChangeText={username => this.handleUsernameChange(username)} onEnterKeyDown={() => this.handleContinue()} />
+        {this._usernameTips()}
```
Comment (chrisnojima) [inline shared/profile/prove-enter-username.desktop.js]: might be better to make a stateless component. feels like nicer to do
```
<UsernameTips platform={platform} />
```

**Comment by keybase-ci-visdiff** [comment]
:mag_right: The commits 1fb175743dac...6e76470d0fea introduced some visual changes on linux:
 * Added: **New_Proof_Enter_Username-Website**  
   New_Proof_Enter_Username-Website rendered
 * Added: **New_Proof_Enter_Username-Twitter**  
   New_Proof_Enter_Username-Twitter rendered
 * Added: **New_Proof_Enter_Username-Reddit**  
   New_Proof_Enter_Username-Reddit rendered
 * Added: **New_Proof_Enter_Username-Hacker_News**  
   New_Proof_Enter_Username-Hacker_News rendered
 * Added: **New_Proof_Enter_Username-Git_Hub**  
   New_Proof_Enter_Username-Git_Hub rendered
 * Added: **New_Proof_Enter_Username-DNS**  
   New_Proof_Enter_Username-DNS rendered
 * Added: **New_Proof_Enter_Username-Coinbase**  
   New_Proof_Enter_Username-Coinbase rendered
 * Added: **New_Proof_Enter_Username-Bitcoin_Disabled** [(view)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/New_Proof_Enter_Username-Bitcoin_Disabled-6e76470d0fea.png)
 * Added: **New_Proof_Enter_Username-Bitcoin** [(view)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/New_Proof_Enter_Username-Bitcoin-6e76470d0fea.png)
 * Added: **New_Proof_Confirm_or_Pending-Pending_dns** [(view)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/New_Proof_Confirm_or_Pending-Pending_dns-6e76470d0fea.png)
 * Added: **New_Proof_Confirm_or_Pending-Pending_Hacker_News** [(view)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/New_Proof_Confirm_or_Pending-Pending_Hacker_News-6e76470d0fea.png)
 * Added: **New_Proof_Confirm_or_Pending-Confirm_http** [(view)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/New_Proof_Confirm_or_Pending-Confirm_http-6e76470d0fea.png)
 * Added: **New_Proof_Confirm_or_Pending-Confirm_Twitter** [(view)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/New_Proof_Confirm_or_Pending-Confirm_Twitter-6e76470d0fea.png)
 * Added: **New_Proof_Confirm_or_Pending-Confirm_Reddit** [(view)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/New_Proof_Confirm_or_Pending-Confirm_Reddit-6e76470d0fea.png)
 * Added: **New_Proof_Confirm_or_Pending-Confirm_Github** [(view)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/New_Proof_Confirm_or_Pending-Confirm_Github-6e76470d0fea.png)
 * Added: **New_Proof_Confirm_or_Pending-Confirm_Coinbase** [(view)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/New_Proof_Confirm_or_Pending-Confirm_Coinbase-6e76470d0fea.png)
 * Added: **New_Proof_Confirm_or_Pending-Confirm_Bitcoin** [(view)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/New_Proof_Confirm_or_Pending-Confirm_Bitcoin-6e76470d0fea.png)
 * Removed: **My_Profile_Confirm_or_Pending-Pending_dns** [(view original)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/My_Profile_Confirm_or_Pending-Pending_dns-1fb175743dac.png)
 * Removed: **My_Profile_Confirm_or_Pending-Pending_Hacker_News** [(view original)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/My_Profile_Confirm_or_Pending-Pending_Hacker_News-1fb175743dac.png)
 * Removed: **My_Profile_Confirm_or_Pending-Confirm_http** [(view original)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/My_Profile_Confirm_or_Pending-Confirm_http-1fb175743dac.png)
 * Removed: **My_Profile_Confirm_or_Pending-Confirm_Twitter** [(view original)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/My_Profile_Confirm_or_Pending-Confirm_Twitter-1fb175743dac.png)
 * Removed: **My_Profile_Confirm_or_Pending-Confirm_Reddit** [(view original)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/My_Profile_Confirm_or_Pending-Confirm_Reddit-1fb175743dac.png)
 * Removed: **My_Profile_Confirm_or_Pending-Confirm_Github** [(view original)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/My_Profile_Confirm_or_Pending-Confirm_Github-1fb175743dac.png)
 * Removed: **My_Profile_Confirm_or_Pending-Confirm_Coinbase** [(view original)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/My_Profile_Confirm_or_Pending-Confirm_Coinbase-1fb175743dac.png)
 * Removed: **My_Profile_Confirm_or_Pending-Confirm_Bitcoin** [(view original)](https://keybase-app-visdiff.s3.amazonaws.com/1468961811947-1fb175743dac-6e76470d0fea-linux/My_Profile_Confirm_or_Pending-Confirm_Bitcoin-1fb175743dac.png)

**Comment by MarcoPolo** [comment]
👍

File: shared/profile/prove-enter-username.desktop.js (hunk 1 above)
Comment (chromakode) [inline shared/profile/prove-enter-username.desktop.js]: Good idea! Done :)

File: shared/profile/prove-enter-username.desktop.js (hunk 2 above)
Comment (chromakode) [inline shared/profile/prove-enter-username.desktop.js]: Good call, that is a lot nicer. Changed.

**Pull request merged**